WaterLog_Backend/Procedures.cs

```diff
@@ -398,8 +419,9 @@ public async Task<double> CalculateTotalWastageLitres(SegmentLeaksEntry leak)
             .Where(inlist => inlist.SegmentsId == leak.SegmentsId)
             .Last();
 
-            var timebetween = (leak.LatestTimeStamp - leak.OriginalTimeStamp).TotalHours;
-            var perhour = CalculatePerHourWastageLitre(leak);
+            var timebetween = (leak.LatestTimeStamp - leak.OriginalTimeStamp).TotalMinutes;
+            timebetween /= 60.00;
+            var perhour = await CalculatePerHourWastageLitre(leak);
```

Comment:
You are using this function quite a bit `CalculatePerHourWastageLitre` please make sure either that function have proper error handling to ensure that correct values are passed back, or check the values when it returns to ensure that you don't get unpredicted behaviour when you do calculations with the variables

Reply:
We are using a Math.Max function to return 0 if any value falls into a negative.